### PR TITLE
Adding partner colours (word, pdf, onenote, etc) for file content attachments

### PR DESCRIPTION
**Pull request by leonardjoe**
# Guide

**This "Help" section can be deleted before submitting this pull request.**

*Update the name of this pull request to reflect the following shape:*

```
{type}/{scope?}/{message}
```

* **type** - A [conventional commit type](https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional#type-enum) **REQUIRED**
* **scope** - The kabab-case scope of the changes in this request
* **message** - A short, kebab-case statement describing the changes **REQUIRED**

Adding new tokens to support "partner" colours for file bricklet attachments (word, excel, PDF, etc).

* [Readme](https://github.com/momentum-design/tokens/blob/master/README.md)
* [Getting Started Guide](https://github.com/momentum-design/tokens/blob/master/GETTING_STARTED.md)
* [Contributing Guide](https://github.com/momentum-design/tokens/blob/master/CONTRIBUTING.md)

# Description

https://www.figma.com/file/n9T5usku57ecvPArT9SpWO/Tokens---Core-Color?node-id=3906%3A2&t=ZYuvVMkkHBjPHmSI-0
https://www.figma.com/file/H19CutixoN8SJGO5EXwcc8/Components---MacOS?node-id=2547%3A0&t=oJe9sbjQooTxKrJG-0

Screenshot 2023-01-11 at 15 03 15
Screenshot 2023-01-11 at 15 04 35

# Links

*Links to relevent resources.*

### CODE REVIEW

File: core/color/partners.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (leonardjoe) [inline core/color/partners.json:1]: these are the colours required, as defined here: https://www.figma.com/file/n9T5usku57ecvPArT9SpWO/Tokens---Core-Color?node-id=3906%3A2&t=ZYuvVMkkHBjPHmSI-0

File: platformcomponents/desktop/attachment-file-branded.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{
+  "attachment-file-branded": {
```
Comment (leonardjoe) [inline platformcomponents/desktop/attachment-file-branded.json:2]: I based this on the existing ```attachment-file-default.json``` which covers the "default" (non-partner) files

File: theme-data/win-system-hc/win-system-hc-common.json [hunk 3]
```diff
@@ -143,6 +143,17 @@
           "attention": "@color-hc-PlaceholderColor",
           "secure": "@color-hc-PlaceholderColor"
         }
+      },
+      "content": {
```
Comment (leonardjoe) [inline theme-data/win-system-hc/win-system-hc-common.json:147]: not sure if this is the correct thing to do, but the script (```npm run build:windows```) gave me errors until I added this definition in here also

**Review by NotNestor — APPROVED** [review]

File: platformcomponents/desktop/attachment-file-branded.json [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    "sent": {
+      "comment": "Applied to sent & safe branded file attachments",
+      "figma": "https://www.figma.com/file/NaNrfXjygZtRgMfHAFHjsp/Components---Windows?node-id=5428%3A29",
+      "border": "@theme-outline-secondary-normal",
+      "previewBox": {
+        "icon": "@theme-common-text-white",
+        "background-pdf": "@theme-common-content-content-pdf",
```
Comment (nataliadelmar) [inline platformcomponents/desktop/attachment-file-branded.json:9]: let's do a parent for background that has pdf, word, etc.. inside

File: theme-data/common/common.json [hunk 5]
```diff
@@ -132,6 +132,17 @@
           "attention": "@color-red-40",
           "secure": "@color-blue-40"
         }
+      },
+      "content": {
+        "content-pdf": "@color-partner-pdf",
```
Comment (nataliadelmar) [inline theme-data/common/common.json:137]: let's not repeat 'content' in the key name of lines 137 - 145

File: theme-data/win-system-hc/win-system-hc-common.json [hunk 6]
```diff
@@ -143,6 +143,17 @@
...
+        "content-pdf": "@color-partner-pdf",
```
Comment (nataliadelmar) [inline theme-data/win-system-hc/win-system-hc-common.json:148]: You get the values from here. In this case, all content colors would be @color-hc-SystemColorHighlightColor

"pdf": "@color-hc-SystemColorHighlightColor"
"word": "@color-hc-SystemColorHighlightColor"
"powerpoint": "@color-hc-SystemColorHighlightColor"
...
"keynote": "@color-hc-SystemColorHighlightColor"

**Review by nataliadelmar — CHANGES_REQUESTED** [review]

**Review by nataliadelmar — APPROVED** [review]

**Pull request merged**